tests/Tests.AzureAppConfiguration/TagsFilterTests.cs

```diff
@@ -0,0 +1,496 @@
+
+                CreateConfigurationSetting("TestKey4", "label", "TestValue4", "bb203f2b-c113-44fc-995d-b933c2143340",
+                    new Dictionary<string, string> { { "Environment", "Staging" }, { "App", "TestApp" }, { "Component", "Frontend" } }),
+
+                CreateConfigurationSetting("TestKey5", "label", "TestValue5", "bb203f2b-c113-44fc-995d-b933c2143341",
+                    new Dictionary<string, string> { { "Special:Tag", "Value:With:Colons" }, { "Tag@With@At", "Value@With@At" } }),
+
+                CreateConfigurationSetting("TestKey6", "label", "TestValue6", "bb203f2b-c113-44fc-995d-b933c2143342",
+                    new Dictionary<string, string> { { "Tag,With,Commas", "Value,With,Commas" }, { "Simple", "Tag" } }),
+
+                CreateFeatureFlagSetting("Feature1", "label", true, "0a76e3d7-7ec1-4e37-883c-9ea6d0d89e63",
+                    new Dictionary<string, string> { { "Environment", "Development" }, { "App", "TestApp" } }),
+
+                CreateFeatureFlagSetting("Feature2", "label", false, "31c38369-831f-4bf1-b9ad-79db56c8b989",
+                    new Dictionary<string, string> { { "Environment", "Production" }, { "App", "TestApp" } }),
+
+                CreateFeatureFlagSetting("Feature3", "label", true, "bb203f2b-c113-44fc-995d-b933c2143339",
+                    new Dictionary<string, string> { { "Environment", "Development" }, { "Component", "API" } }),
+
+                CreateFeatureFlagSetting("Feature4", "label", false, "bb203f2b-c113-44fc-995d-b933c2143340",
+                    new Dictionary<string, string> { { "Environment", "Staging" }, { "App", "TestApp" }, { "Component", "Frontend" } }),
+
+                CreateFeatureFlagSetting("Feature5", "label", true, "bb203f2b-c113-44fc-995d-b933c2143341",
+                    new Dictionary<string, string> { { "Special:Tag", "Value:With:Colons" }, { "Tag@With@At", "Value@With@At" } }),
+
+                CreateFeatureFlagSetting("Feature6", "label", false, "bb203f2b-c113-44fc-995d-b933c2143342",
+                    new Dictionary<string, string> { { "Tag,With,Commas", "Value,With,Commas" }, { "Simple", "Tag" } })
+            };
+        }
+
+        private ConfigurationSetting CreateConfigurationSetting(string key, string label, string value, string etag, IDictionary<string, string> tags)
+        {
+            // Create the setting without tags
+            var setting = ConfigurationModelFactory.ConfigurationSetting(
+                key: key,
+                label: label,
+                value: value,
+                eTag: new ETag(etag),
+                contentType: "text");
+
+            // Add tags to the setting
+            if (tags != null)
+            {
+                foreach (var tag in tags)
+                {
+                    setting.Tags.Add(tag.Key, tag.Value);
+                }
+            }
+
+            return setting;
+        }
```

Comment:
Can you add a test for tag filter with null/empty value and a test for more than 5 tag filters.

Reply:
I was thinking about how to test more than 5 tag filters. Does it make sense to validate this in the provider like the other basic validation for tags? Otherwise, I can still add a test that will just mock the `RequestFailedException` returned

Reply:
We can add basic validation to throw if user provides more than 5 tag filters

Reply:
Talked to jimmy offline, seems like it would be better to not add this validation since we don't validate this kind of thing in other APIs and the error message that comes from the exception is already sufficient. We don't need to restrict the tag filters count from the provider side

Reply:
I'd forgotten we allow comma separated key filters too and dont do this validation for key filters. So it makes sense to ski the validation for tags also